src/WebJobs.Script/Host/ScriptHost.cs

```diff
@@ -530,6 +516,30 @@ private void AddFunctionDescriptors()
                 _logger.AddingDescriptorProviderForLanguage(_workerRuntime);
                 _descriptorProviders.Add(new RpcFunctionDescriptorProvider(this, _workerRuntime, ScriptOptions, _bindingProviders, _functionDispatcher, _loggerFactory, _applicationLifetime));
             }
+
+            // Codeless functions run side by side with regular functions.
+            // In addition to descriptors already added here, we need to ensure all codeless functions
+            // also have associated descriptors.
+            AddCodelessDescriptors(functionMetadata);
+        }
+
+        /// <summary>
+        /// Adds a DotNetFunctionDescriptorProvider to the list of descriptors if any function metadata has language set to "codeless" in it.
+        /// </summary>
+        private void AddCodelessDescriptors(IEnumerable<FunctionMetadata> functionMetadata)
+        {
+            if (functionMetadata.Any(m => m is ProxyFunctionMetadata))
+            {
+                _descriptorProviders.Add(new ProxyFunctionDescriptorProvider(this, ScriptOptions, _bindingProviders, _loggerFactory));
+            }
+
+            // If we have a non-proxy codeless function, we need to add a .NET descriptor provider. But only if it wasn't already added.
+            // At the moment, we are assuming that all codeless functions will have language as DotNetAssembly
+            if (functionMetadata.Any(m => m.IsCodeless && !(m is ProxyFunctionMetadata))
+                && !_descriptorProviders.Any(d => d is DotNetFunctionDescriptorProvider))
```

Comment:
nit: make the DotNetFunctionDescriptorProvider check first to avoid work in some cases